### PR TITLE
Replace random

### PR DESCRIPTION
**Pull request by jolting**
**Base** 
*    Remove TBB
*    Replace mt19937 implementation

I acknowledge to have: 
* Read the [`CONTRIBUTING`](https://github.com/MRPT/mrpt/blob/master/.github/CONTRIBUTING.md) page
* Updated [`docs/doxygen-pages/changelog.h`](https://github.com/MRPT/mrpt/blob/master/doc/doxygen-pages/changeLog_doc.h) to describe these changes (if applicable)
* Updated [`AUTHORS`](https://github.com/MRPT/mrpt/blob/master/AUTHORS) (if applicable)

(Notify: @MRPT/owners )

### CODE REVIEW

**Comment by jolting** [comment]
Another ~200 lines, not including the cmake file.

**Comment by jlblancoc** [comment]
I'll review this one more carefully just to be sure, but in principle it looks all good, and totally makes sense to rely on the std rnd.:+1:

File: libs/base/src/random/RandomGenerator.cpp [hunk 1]
```diff
@@ -27,76 +27,25 @@ CRandomGenerator mrpt::random::randomGenerator;
...
-inline uint32_t twist( const uint32_t m, const uint32_t s0, const uint32_t s1 ) { return m ^ (mixBits(s0,s1)>>1) ^ (-loBit(s1) & 0x9908b0dfUL); }
-
-#if defined(_MSC_VER)
-	#pragma warning(pop)
-#endif
-
-
 // Generate an array of 624 untempered numbers
 void CRandomGenerator::MT19937_generateNumbers()
 {
```
Comment (jlblancoc) [inline libs/base/src/random/RandomGenerator.cpp]: It's safe now to remove this method entirely

Comment (jolting) [inline libs/base/src/random/RandomGenerator.cpp]: Great!!! I was thinking this was a bit useless with the changes and I should have looked at this closer.

**Comment by jolting** [comment]
Done.
Did you see the new special math routines coming in c++17?

http://en.cppreference.com/w/cpp/numeric/special_math

**Comment by jlblancoc** [comment]
merged, thanks!! :+1: 

> Did you see the new special math routines coming in c++17?

Nope... it has little direct use for this project right now, but it's great to see the available std functions growing!!

**Pull request merged**